src/boosting/gbdt_model_text.cpp

```diff
@@ -632,31 +632,49 @@ std::vector<double> GBDT::FeatureImportance(int num_iteration, int importance_ty
   }
 
   std::vector<double> feature_importances(max_feature_idx_ + 1, 0.0);
+  bool warn_about_feature_number = false;
+  int max_feature_index_found = -1;
   if (importance_type == 0) {
     for (int iter = 0; iter < num_used_model; ++iter) {
       for (int split_idx = 0; split_idx < models_[iter]->num_leaves() - 1; ++split_idx) {
         if (models_[iter]->split_gain(split_idx) > 0) {
+          const int real_feature_index = models_[iter]->split_feature(split_idx);
 #ifdef DEBUG
-          CHECK_GE(models_[iter]->split_feature(split_idx), 0);
+          CHECK_GE(real_feature_index, 0);
 #endif
-          feature_importances[models_[iter]->split_feature(split_idx)] += 1.0;
+          if (static_cast<size_t>(real_feature_index) >= feature_importances.size()) {
+            warn_about_feature_number = true;
```

Comment:
Provide warning instead of enlarging feature importance vector. Since `GBDT::FeatureImportance` is used by Python API and the python code doesn't know the final size of feature importance vector, dynamically enlarging the vector will cause out of bound access.